src/websockets/legacy/server.py

```diff
@@ -640,7 +640,7 @@ async def handshake(
 
         response_headers.setdefault("Date", email.utils.formatdate(usegmt=True))
         if self.server_header is not None:
-            response_headers.setdefault("Server", self.server_header)
+            response_headers["Server"] = self.server_header
```

Comment:
You cannot do this: it's a backwards incompatible change.

Reply:
Oh, so only way is to use `setdefault`?

I assume `response_headers.update({"Server": self.server_header})` will cause backward incompatibility..

Reply:
I'm leaning towards fixing the `Headers.setdefault` method.

Reply:
But I'd have to look at the implementation of the parent class to check that we aren't headed down a rabbit hole if we do that...

Reply:
I am not sure if `Headers` overrides `setdefault`. It's a method of MutableMapping I suppose? (I might be wrong)

Reply:
Just a question out of curiosity, if it's okay.. 

What's the backwards incompatibility here? Aren't we setting value to the Mapping object in both cases?

Reply:
The backwards-incompatibility in your proposal is that you are overriding any previously set value. Using `setdefault` here was an escape hatch to support customizing the `Server` header before we formalized an API to do that (in the change we made together recently).

Reply:
Thanks for the reply!

I found while testing that when we set headers directly (like I did in this PR), it doesn't replace, rather append another header to it. That is, if we have `Server` header already and if user sets another, we will end up having 2 of those.

Reply:
I rearranged and made setting of server header before setting extra headers. This handles the scenario well, without making any change. 

Is this a better solution?

Reply:
>If you set this header (by overriding private APIs, probably...) websockets doesn't override it.

 Another observation:

 If I do this 
```python
        if self.server_header is not None:
            response_headers.setdefault("Server", self.server_header)
            response_headers["Server"] = "Potato"
```
we get
```
Headers([('Upgrade', 'websocket'), ('Connection', 'Upgrade'), ('Sec-WebSocket-Accept', '9i+pz2BTKb6LxQmkX4v7j7g71ps='), ('Server', 'Python/3.10 websockets/10.4.dev47+g99c8440.dirty'), ('Server', 'Potato'), ('Date', 'Mon, 12 Sep 2022 16:14:46 GMT')])
```
Which means doing ` response_headers["Server"] = "Potato"` adds another header, and not override right?

Reply:
> I rearranged and made setting of server header before setting extra headers. This handles the scenario well, without making any change.

> Is this a better solution?

I don't think so: you're breaking the code of those who set `Server` via `extra_headers`.

Reply:
Changing `Headers.setdefault` isn't possible because this methods returns the value and we don't want to do this when there are multiple values.

Reply:
> Changing `Headers.setdefault` isn't possible because this methods returns the value and we don't want to do this when there are multiple values.

Agreed. It'd affect the normal behavior.